StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/Link/LinkURLGeneratorTests.swift

```diff
@@ -0,0 +1,96 @@
+    }
+
+    func testURLCreationRegularUnicode() {
+        var params = testParams
+        params.customerInfo.email = "유니코드"
+        _ = try! LinkURLGenerator.url(params: params)
+        // Just make sure it doesn't fail
+    }
+
+    func testURLCreationHorribleUnicode() {
+        var params = testParams
+        params.customerInfo.email = String(bytes: [0xD8, 0x00] as [UInt8], encoding: .utf16BigEndian)! // Unpaired UTF-16 surrogates
+        do {
+            _ = try LinkURLGenerator.url(params: params)
+            XCTFail("Encoding should fail for invalid data")
+        } catch {
+            XCTAssertTrue(error is EncodingError)
+        }
+    }
+
+    func testURLParamsFromConfig() async {
+        let config = PaymentSheet.Configuration()
+        let intentConfig = PaymentSheet.IntentConfiguration(mode: .payment(amount: 100, currency: "USD")) { _, _, _ in
+            // Nothing
+        }
+        config.apiClient.publishableKey = "pk_123"
+        let intent = Intent.deferredIntent(elementsSession: STPElementsSession.emptyElementsSession, intentConfig: intentConfig)
+        let params = try! await LinkURLGenerator.linkParams(configuration: config, intent: intent)
+
+        let expectedParams = LinkURLParams(paymentObject: .link_payment_method,
+                                           publishableKey: config.apiClient.publishableKey!,
+                                           paymentUserAgent: PaymentsSDKVariant.paymentUserAgent,
+                                           merchantInfo: LinkURLParams.MerchantInfo(businessName: "xctest", country: "US"),
+                                           customerInfo: LinkURLParams.CustomerInfo(country: "US", email: nil),
+                                           paymentInfo: LinkURLParams.PaymentInfo(currency: "USD", amount: 100),
+                                           experiments: [:],
+                                           flags: [:],
+                                           loggerMetadata: [:],
+                                           locale: Locale.init(identifier: "en_US").toLanguageTag())
+
+        XCTAssertEqual(params, expectedParams)
+    }
+}
+
+extension STPElementsSession {
+    // Should use the one from StripeiOSTests, but we don't have good infrastructure to share these
+    // and we're not using any details from it.
+    static var emptyElementsSession: STPElementsSession {
+        let apiResponse: [String: Any] = ["payment_method_preference": ["ordered_payment_method_types": ["123"],
+                                                                        "country_code": "US", ] as [String: Any],
```

Comment:
I changed the country_code to "FR", thinking that currency would come up as "EUR", but it failed ... I don't fully understand why.  Is this expected?

Reply:
There's no logic here, it's just an empty STPElementsSession to include some basic values when testing `LinkURLGenerator.linkParams()` above. Changing one value shouldn't modify the other passed-in values.